drivers/flash/flash_stm32l4x.c

```diff
@@ -149,9 +149,15 @@ int flash_stm32_write_range(struct device *dev, unsigned int offset,
 			    const void *data, unsigned int len)
 {
 	int i, rc = 0;
+	u32_t *tmp = (u32_t *)data;
+	u64_t dword;
 
 	for (i = 0; i < len; i += 8, offset += 8) {
-		rc = write_dword(dev, offset, ((const u64_t *) data)[i>>3]);
+		/* Use local variable to access 'data' word by word */
+		/* to avoid errors in case 'data' is unaligned */
+		dword = ((u64_t)(tmp + (i >> 2))[1] << 32) +
```

Comment:
Would UNALIGNED_GET be useful here?

Reply:
Let me have a test

Reply:
Seems like UNALIGNED_GET doesn't like the expression it should be applied to.

Reply:
`UNALIGNED_GET` needs a pointer in argument, so you need to use UNALIGNED_GET((const u64_t *) data + (i >> 3))``

Reply:
Right! fixed,  thanks again.